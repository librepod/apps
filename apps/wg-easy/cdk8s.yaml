language: typescript
app: bun run main.ts
