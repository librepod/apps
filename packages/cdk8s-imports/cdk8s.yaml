language: typescript
app: npx ts-node main.ts
imports:
  - k8s
  - ./argoproj.io_crd.yaml
