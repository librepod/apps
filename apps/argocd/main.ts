import { Construct } from 'constructs'
import { App, Chart, ChartProps } from 'cdk8s'
import { ArgocdApplication } from '@repo/cdk8s-imports/imports/argoproj.io'

export class LibrePodApps extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props)

    new ArgocdApplication(this, id, {
      metadata: {
        finalizers: ['resources-finalizer.argocd.argoproj.io'],
        annotations: {
          'argocd.argoproj.io/sync-wave': '0'
        },
        name: 'librepod-apps',
        namespace: 'argocd'
      },
      spec: {
        ignoreDifferences: [
          {
            group: 'argoproj.io',
            jsonPointers: ['/status'],
            kind: 'Application'
          }
        ],
        project: 'default',
        source: {
          repoUrl: 'https://github.com/librepod/apps',
          targetRevision: 'HEAD',
          path: 'apps/wg-easy',
          plugin: {
            name: 'cdk8s'
          }
        },
        destination: {
          namespace: 'wg-easy',
          server: 'https://kubernetes.default.svc'
        },
        syncPolicy: {
          syncOptions: [ 'CreateNamespace=true' ],
          automated: {
            allowEmpty: true,
            prune: true,
            selfHeal: true
          }
        }
      }
    })
  }
}

const app = new App()
new LibrePodApps(app, 'librepod-apps', { disableResourceNameHashes: true })
app.synth()
