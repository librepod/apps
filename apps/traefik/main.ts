import { name as appName, version } from './package.json'
import { Construct } from 'constructs'
import { Helm, App, Chart, ChartProps } from 'cdk8s'
import * as utils from '@repo/cdk8s-utils'

export class MyChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props)

    new Helm(this, appName, {
      repo: 'https://traefik.github.io/charts',
      chart: appName,
      namespace: appName,
      releaseName: appName,
      version: version,
      values: utils.parseValues('values.yaml')
    });
  }
}

const app = new App()
new MyChart(app, appName, { disableResourceNameHashes: true })
app.synth()
