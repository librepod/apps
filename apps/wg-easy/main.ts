import { Construct } from 'constructs'
import { App, Chart, ChartProps } from 'cdk8s'

import createDeployment from './lib/deployment'

export class WgEasy extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props)

    createDeployment(this)
  }
}

const app = new App()
new WgEasy(app, 'wg-easy', { disableResourceNameHashes: true })
app.synth()
