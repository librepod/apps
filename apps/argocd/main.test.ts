import {LibrePodApps} from './main';
import {Testing} from 'cdk8s';

describe('Placeholder', () => {
  test('Empty', () => {
    const app = Testing.app();
    const chart = new LibrePodApps(app, 'test-chart');
    const results = Testing.synth(chart)
    expect(results).toMatchSnapshot();
  });
});
