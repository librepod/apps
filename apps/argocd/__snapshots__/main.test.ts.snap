// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder Empty 1`] = `
Array [
  Object {
    "apiVersion": "argoproj.io/v1alpha1",
    "kind": "Application",
    "metadata": Object {
      "annotations": Object {
        "argocd.argoproj.io/sync-wave": "0",
      },
      "finalizers": Array [
        "resources-finalizer.argocd.argoproj.io",
      ],
      "name": "librepod-apps",
      "namespace": "argocd",
    },
    "spec": Object {
      "destination": Object {
        "namespace": "argocd",
        "server": "https://kubernetes.default.svc",
      },
      "ignoreDifferences": Array [
        Object {
          "group": "argoproj.io",
          "jsonPointers": Array [
            "/status",
          ],
          "kind": "Application",
        },
      ],
      "project": "default",
      "source": Object {
        "path": "apps/wg-easy",
        "plugin": Object {
          "name": "cdk8s",
        },
        "repoURL": "https://github.com/librepod/apps",
        "targetRevision": "HEAD",
      },
      "syncPolicy": Object {
        "automated": Object {
          "allowEmpty": true,
          "prune": true,
          "selfHeal": true,
        },
      },
    },
  },
]
`;
